### PR TITLE
Sync humanoid with client when changing scene.

### PR DESCRIPTION
**Pull request by 0mdc**
This syncs the VR controller with the humanoid position when changing scene.

Depends on: https://github.com/facebookresearch/habitat-lab/pull/1623

### CODE REVIEW

File: Assets/Scripts/GfxReplayPlayer.cs [hunk 1]
```diff
@@ -18,6 +19,11 @@ void Awake()
         {
             Debug.LogWarning($"Highlight manager missing from '{name}'. Object highlights will be ignored.");
         }
+        _playerPositionHandler = GetComponent<PlayerPositionHandler>();
+        if (_highlightManager == null)
```
Comment (eundersander) [inline Assets/Scripts/GfxReplayPlayer.cs]: This looks like a bug

File: Assets/Scripts/PlayerPositionHandler.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        if (!enabled) return;
+
+        var humanoidPosition = keyframe?.message.humanoidPosition;
+        if (humanoidPosition != null && humanoidPosition.Count == 3)
+        {
+            Vector3 newPosition = CoordinateConventionHelper.ToUnityVector(humanoidPosition);
+
+            Vector3 delta = newPosition - xrCamera.transform.position;
+            delta = new Vector3(delta.x, 0.0f, delta.z); // Ignore y-axis delta
```
Comment (eundersander) [inline Assets/Scripts/PlayerPositionHandler.cs]: It's fine to ignore the y here. In the future, for multi-story houses, we'll have to find a way to adjust height, too (presumably Quest somehow exposes the headset's height above the physical ground).

**Review by eundersander — APPROVED** [review]
LGTM!

File: Assets/Scripts/PlayerPositionHandler.cs (hunk 2 above)
Comment (0mdc) [reply Assets/Scripts/PlayerPositionHandler.cs]: Indeed, great point.

**Pull request merged**